presto-orc/src/main/java/com/facebook/presto/orc/writer/SliceDictionaryColumnWriter.java

```diff
@@ -91,10 +91,10 @@ public int getDictionaryEntries()
     }
 
     @Override
-    protected boolean tryConvertRowGroupToDirect(int dictionaryIndexCount, int[] dictionaryIndexes, int maxDirectBytes)
+    protected boolean tryConvertRowGroupToDirect(int dictionaryIndexCount, int[] dictionaryIndexes, int maxDirectBytes, Optional<ColumnStatistics> statistics)
     {
+        directColumnWriter.beginRowGroup();
```

Comment:
begin and finishRowGroup here should be orthogonal from stats reuse. Any better abstraction we can think of to avoid the implication?

Reply:
Here is my thinking: previously tryConvertToDirect was responsible for calling the beginRowGroup and finishRowGroup. After this change each rowGroup write itself will be responsible. Presence of statistics means that, rowGroup is complete and finishRowGroup should be called. If statistics does not exist, then it is a partial row group and finishRowGroup should not be called. But beginRowGroup happens unconditionally.